.github/workflows/deploy-python-backend-dev.yml

```diff
@@ -0,0 +1,36 @@
+name: Deploy the frontend and backend to the Firebase Production environment.
```

Comment:
nit: We should probably rename this to distinguish it from the other Firebase deploy function - they both show up with the same name in the GitHub UI. Maybe "Deploy Python functions to the Firebase Dev environment" or "Deploy Python Functions - DEV"? @chiroptical